cpp/src/strings/copying/copying.cu

```diff
@@ -42,6 +44,38 @@ std::unique_ptr<cudf::column> copy_slice(strings_column_view const& strings,
   if (end < 0 || end > strings_count) end = strings_count;
   CUDF_EXPECTS(((start >= 0) && (start < end)), "Invalid start parameter value.");
   strings_count = cudf::util::round_up_safe<size_type>((end - start), step);
+  if (start == 0 && strings.offset() == 0) {
+    // slice starts at the beginning, so do not need a gather to adjust offsets
+    auto offsets_column = make_numeric_column(
```

Comment:
The `gather` call is really only necessary for `step > 1`.
This whole thing probably can be simplified with something like the following:
```
if( start==0 && strings.offset()==0 && step==1 ) {
  auto offsets_column = std::make_unique<cudf::column>( 
     cudf::slice(strings.offsets(),{0,strings_count+1}).front()),
     stream, mr );
  auto data_size = cudf::detail::get_value<int32_t>(offsets_column->view(), strings_count, stream);
  auto chars_column = std::make_unique<cudf::column>(
     cudf::slice(strings.chars(),{0,data_size}).front(),
     stream, mr );
  return make_strings_column(strings_count,
                             std::move(offsets_column),
                             std::move(chars_column),
                             UNKNOWN_NULL_COUNT,
                             copy_bitmask(strings.null_mask(), 0, strings_count, stream, mr),
                             stream,
                             mr);
}
// do the gather instead
```

Reply:
Thanks for the suggestion, this is much cleaner!  Updated.